transport/httptransport/http_transport.go

```diff
@@ -0,0 +1,235 @@
+package httptransport
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"net/url"
+	"os"
+	"sync"
+
+	"github.com/google/uuid"
+
+	logging "github.com/ipfs/go-log/v2"
+
+	"github.com/filecoin-project/boost/transport/types"
+
+	"github.com/filecoin-project/boost/transport"
+)
+
+var log = logging.Logger("http-transport")
+
+const (
+	// 1 Mb
+	readBufferSize = 1048576
+)
+
+var _ transport.Transport = (*httpTransport)(nil)
+
+func New() *httpTransport {
+	return &httpTransport{}
+}
+
+type httpTransport struct {
+	// TODO Construct and inject an http client here ?
+}
+
+func (h *httpTransport) Execute(ctx context.Context, transportInfo []byte, dealInfo *types.TransportDealInfo) (th transport.Handler, err error) {
+	// de-serialize transport opaque token
+	tInfo := &types.HttpRequest{}
+	if err := json.Unmarshal(transportInfo, tInfo); err != nil {
+		return nil, fmt.Errorf("failed to de-serialize transport info bytes: %w", err)
+	}
+	// check that the outputFile exists
+	fi, err := os.Stat(dealInfo.OutputFile)
+	if err != nil {
+		return nil, fmt.Errorf("output file state error: %w", err)
+	}
+	// validate req
+	if _, err := url.Parse(tInfo.URL); err != nil {
+		return nil, fmt.Errorf("failed to parse url: %w", err)
+	}
+	// do we have more bytes than required already ?
+	fileSize := fi.Size()
+	if fileSize > dealInfo.DealSize {
+		return nil, fmt.Errorf("deal size=%d but file size=%d", dealInfo.DealSize, fileSize)
+	}
+
+	// construct the transfer instance that will act has the transfer handler
+	tctx, cancel := context.WithCancel(ctx)
+	t := &transfer{
+		cancel:         cancel,
+		tInfo:          tInfo,
+		dealInfo:       dealInfo,
+		eventCh:        make(chan types.TransportEvent, 256),
+		nBytesReceived: fileSize,
+	}
+
+	// is the transfer already complete ? we check this by comparing the number of bytes
+	// in the output file with the deal size.
+	if fileSize == dealInfo.DealSize {
+		defer close(t.eventCh)
```

Comment:
Should we also call cancel() here? Alternatively, should we call t.Close()?

Reply:
We don't need to call Close as the transfer go-routine hasn't started yet. This check is done when the user calls `Execute` and want to ensure that the file dosen't already have as many bytes as the user needs.

Reply:
Ok makes sense, I think we still want to call cancel() here though

Reply:
Good catch.